main.tf

```diff
@@ -177,12 +176,63 @@ resource "github_branch_default" "default" {
 }
 
 # ---------------------------------------------------------------------------------------------------------------------
-# Branch Protection
+# v4 Branch Protection
+# https://registry.terraform.io/providers/integrations/github/latest/docs/resources/branch_protection
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "github_branch_protection" "this" {
+  for_each = var.branch_protections_v4
```

Comment:
I've been able to support v3 & v4 side by side using the following check here : `for_each = length(coalesce(var.branch_protections_v4, {})) != 0 ? var.branch_protections_v4 : {}`